src/Views/VPNPage.vala

```diff
@@ -112,128 +93,65 @@ public class Network.VPNPage : Network.Widgets.Page {
         list_root.attach (scrolled, 0, 0, 1, 1);
         list_root.attach (toolbar, 0, 1, 1, 1);
 
-        var main_frame = new Gtk.Frame (null);
-        main_frame.vexpand = true;
-        main_frame.get_style_context ().add_class (Gtk.STYLE_CLASS_VIEW);
-        main_frame.add (list_root);
+        var frame = new Gtk.Frame (null);
+        frame.vexpand = true;
+        frame.get_style_context ().add_class (Gtk.STYLE_CLASS_VIEW);
+        frame.add (list_root);
 
-        content_area.row_spacing = 12;
-        content_area.add (top_revealer);
-        content_area.add (main_frame);
+        var main_overlay = new Gtk.Overlay ();
+        main_overlay.add (frame);
+        main_overlay.add_overlay (remove_vpn_toast);
 
-        action_area.add (new Network.Widgets.SettingsButton ());
+        content_area.row_spacing = 12;
+        content_area.add (main_overlay);
 
         show_all ();
 
+        vpn_list.row_selected.connect (row => {
```

Comment:
This signal is fired both when a row is selected and when the selection is cleared. We should be careful to check that `row` is not null here: https://valadoc.org/gtk+-3.0/Gtk.ListBox.row_selected.html

Reply:
see https://github.com/elementary/switchboard-plug-network/pull/210